webpush/encrypt.go

```diff
@@ -55,13 +55,43 @@ import (
 	"strings"
 )
 
+// ContentEncoding indicates the version of encoding.
+type ContentEncoding int
+
 const (
-	maxPayloadLength = 4078
+	// The encoding that is widely deployed with WebPush (as of 2016-11).
+	AESGCM ContentEncoding = iota
+	// The most recent encoding, the salt, record size and key identifier
+	// are included in a header that is part of the encrypted content coding.
+	AES128GCM
+)
+
+func (v ContentEncoding) String() string {
+	switch v {
+	case AESGCM:
+		return "aesgcm"
+	case AES128GCM:
+		return "aes128gcm"
+	}
+	panic("A valid content encoding must be used")
```

Comment:
reverted back to original in that case and returning unknown. Also added a check in newCEK and return an error if the encoding is unknown, but it's a bit awkward because they probably can't pass any value other than AES and AES128. Anyway take a look.